src/Client/InternalClient.js

```diff
@@ -1301,129 +1301,89 @@ export default class InternalClient {
-		return this.resolver.resolveChannel(chann)
-		.then(channel => {
-			if (channel.type !== "text") {
-				return Promise.reject(new Error("Channel must be a text channel"));
-			}
+	setChannelNameAndTopic(channel) {
+		name = name || "unnamed-channel";
+		topic = topic || "";
 
-			this.apiRequest("patch", Endpoints.CHANNEL(channel.id), true, {
-				name: name,
-				position: channel.position,
-				topic: topic
-			})
-			.then(res => {
-				channel.name = res.name;
-				channel.topic = res.topic;
-			})
-		});
+		return this.updateChannel(channel, {name: name, topic: topic});
 	}
 
 	//def setChannelPosition
-	setChannelPosition(chann, position = 0) {
-		return this.resolver.resolveChannel(chann)
-		.then(channel => {
-			this.apiRequest("patch", Endpoints.CHANNEL(channel.id), true, {
-				name: channel.name,
-				position: position,
-				topic: channel.topic,
-				user_limit: channel.userLimit,
-				bitrate: channel._bitrate
-			})
-			.then(res => channel.position = res.position)
-		});
+	setChannelPosition(channel, position) {
+		position = position || 0;
+
+		return this.updateChannel(channel, {position: position});
 	}
 
 	//def setChannelUserLimit
 	setChannelUserLimit(channel, limit) {
 		limit = limit || 0; // default 0 = no limit
 
-		if (limit < 0) {
-			return Promise.reject(new Error("User limit cannot be less than 0"));
-		}
-
-		if (limit > 99) {
-			return Promise.reject(new Error("User limit cannot be greater than 99"));
-		}
-
-		return this.resolver.resolveChannel(channel).then((channel) => {
-			if (channel.type !== "voice") {
-				return Promise.reject(new Error("Channel must be a voice channel"));
-			}
-
-			return this.apiRequest("patch", Endpoints.CHANNEL(channel.id), true, {
-				name: channel.name,
-				position: channel.position,
-				user_limit: limit,
-				bitrate: channel._bitrate
-			})
-			.then(res => channel.userLimit = limit);
-		});
+		return this.updateChannel(channel, {userLimit: limit})
 	}
 
 	//def setChannelBitrate
 	setChannelBitrate(channel, kbitrate) {
 		kbitrate = kbitrate || 64; // default 64kbps
 
-		if (kbitrate < 8 || kbitrate > 96) {
-			return Promise.reject(new Error("Bitrate must be between 8-96kbps"));
-		}
+		return this.updateChannel(channel, {bitrate: kbitrate});
+	}
 
+	//def updateChannel
+	updateChannel(channel, data) {
 		return this.resolver.resolveChannel(channel).then(channel => {
-			if (channel.type !== "voice") {
-				return Promise.reject(new Error("Channel must be a voice channel"));
+			if (!channel) {
+				return Promise.reject(new Error("Failed to resolve channel"));
 			}
 
-			return this.apiRequest("patch", Endpoints.CHANNEL(channel.id), true, {
-				name: channel.name,
-				user_limit: channel.userLimit,
-				position: channel.position,
-				bitrate: kbitrate * 1000 // in bps
-			})
+			data = {
+				name: data.name || channel.name,
+				topic: data.topic || channel.topic,
+				position: data.position || channel.position,
```

Comment:
You have to be careful with that syntax with bools and numbers
`position: 0 || 3` will return `3`
`user_limit: 0 || 12` will return `12`